electrode-ota-server-model-app/src/app.js

```diff
@@ -523,7 +580,29 @@ export default (options, dao, upload, logger) => {
     getMetricsForDeployment(deployment) {
       return this.getMetricsFromCache(deployment).then(result => {
         if (result && result.summaryJson) {
-          return JSON.parse(result.summaryJson);
+          const summary = JSON.parse(result.summaryJson);
+          // fetch latest metrics data
+          return dao.metricsByStatusAndTime(
+            deployment.key, new Date(result.lastRunTimeUTC), new Date(Date.now())
+          ).then((metrics = []) => {
+            const { label } = deployment.package || {};
+            logger.info({ deployment: deployment.name }, "fetched latest metrics by status and time");
+            // cleanup and merge the results
+            const latestSummary = doSummarize(metrics, label);
+            if (isValidObject(latestSummary)) {
+              for (const k in latestSummary) {
+                if (!summary[k]) {
+                  summary[k] = { active: 0, downloaded: 0, failed: 0, installed: 0 };
+                }
+                summary[k].active += latestSummary[k].active;
+                summary[k].downloaded += latestSummary[k].downloaded;
+                summary[k].failed += latestSummary[k].failed;
+                summary[k].installed += latestSummary[k].installed;
+              }
+            }
+
+            return summary;
+          });
```

Comment:
can u this in another function?

Reply:
u mean to a different file ?

Reply:
another function in same file.

Reply:
moved the metrics merger to as separate method.